wagtail_newsletter/campaign_backends/mailchimp.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from mailchimp_marketing import Client
+from mailchimp_marketing.api_client import ApiClientError
+
+from . import Audience, AudienceNotFoundError, AudienceSegment, CampaignBackend
+
+
+class MailchimpCampaignBackend(CampaignBackend):
+    def __init__(self):
+        self.client = Client()
+        self.client.set_config(self.get_client_config())
+
+    def get_client_config(self):
```

Comment:
[picky mode] since the rest of the class has type hints... Maybe this method could be type-hinted too? :slightly_smiling_face: 

Something like this for example:
```py
if TYPE_CHECKING:
  class ClientConfig(TypedDict):
      api_key: str
      timeout: int

# ...and then:

 def get_client_config(self) -> "ClientConfig":
     ...

Reply:
I've added a simpler typing hint to `get_client_config`, saying just that it returns a `dict[str, Any]`. The Mailchimp library's [`set_config()`](https://github.com/mailchimp/mailchimp-marketing-python/blob/3305fa45b3f436767a539c5fba9cb2b0a083d761/build/lib/mailchimp_marketing/api_client.py#L51-L66) method accepts a couple more keys in the dict, and more could be added, and I don't think it's worth mapping them in our code TBH.